utils/utils.c

```diff
@@ -209,14 +228,25 @@ int create_directory(char *dirname)
 	}
 
 	if (!access(dirname, F_OK)) {
-		if (!access(oldname, F_OK) && remove_directory(oldname) < 0) {
+		if (!access(oldname, F_OK)) {
+			if (is_uftrace_dir(oldname)) {
+				if (remove_directory(oldname) < 0)
+					pr_warn("removing old directory "\
+							"failed: %m\n");
+			}
+			else {
+				pr_warn("%s is not a uftrace data path.\n");
+			}
```

Comment:
Hmm.. I don't like a deeply nested if statements.  What about something like this?

```c
if (access(OK) {
   if (!is_uftrace_dir)
      warn_and_out
   if (remove_directory < 0)
      warn_and_out
}
```